[W14-3] HealthBase

lingzhiyu opened the pull request:
In progress @jeffypie369 @yuntongzhang @snajef @GAOJIAXIN666

kexiaowen commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,52 @@
-Role: Developer +
-Responsibilities: UI
+Role: TBD +
+Responsibilities: TBD +
```
Try to decide on the roles and responsibilities of each member over the recess week and update this section!

kexiaowen replied:
This section is still not updated :(

kexiaowen commented on docs/DeveloperGuide.adoc:
```diff
@@ -902,6 +705,21 @@ Windows, Linux, Unix, OS-X
+Patient is discharged from the hospital
+
 [appendix]
 == Product Survey
```
You can delete this section if you are not doing product survey.

kexiaowen commented on docs/DeveloperGuide.adoc:
```diff
@@ -149,15 +149,15 @@ image::LogicClassDiagram.png[width="800"]
 
 .Component interactions for `delete 1` command (part 1)
-image::SDforDeletePerson.png[width="800"]
+image::SDforDeletepatient.png[width="800"]
```
Image is not rendering. You need to change the name of the image under main/docs/images/ as well.

kexiaowen commented on docs/DeveloperGuide.adoc:
```diff
@@ -149,15 +149,15 @@ image::LogicClassDiagram.png[width="800"]
 
 .Component interactions for `delete 1` command (part 2)
-image::SDforDeletePersonEventHandling.png[width="800"]
+image::SDforDeletepatientEventHandling.png[width="800"]
```
Same issue as above.

kexiaowen reviewed (commented):
Well done guys! 👍Good effort in updating the docs. Keep it up :) You can make use of recess week and each member can try to do a local-impact change to the code base.

@snajef Pls remember to set your commit email address and username in git so that your commits will be reflected on github. You may refer to these links: https://help.github.com/articles/setting-your-commit-email-address-in-git/ 
https://help.github.com/articles/setting-your-username-in-git/

kexiaowen commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,52 @@
-image::yijinl.jpg[width="150", align="left"]
-{empty}[http://github.com/yijinl[github]] [<<johndoe#, portfolio>>]
+=== Ling Zhiyu
+image::omegafishy.png[width="150", align="left"]
```
@omegafishy @jeffypie369 Please change your photos as you should choose a photo that shows your face clearly (i.e. not too small) -- somewhat similar to a passport photo. This was pointed out by prof during TA meeting.

kexiaowen commented on docs/UserGuide.adoc:
```diff
@@ -56,177 +56,211 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+Patient data are saved in the hard disk automatically after any command that changes the data. +
 There is no need to save manually.
 
 // tag::dataencryption[]
```
You can consider adding more features that are tagged as coming in v2.0. For example the authentication/permission system since you mentioned that previously but doesn't seem to implement it

kexiaowen commented on docs/UserGuide.adoc:
```diff
@@ -56,177 +56,211 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-`delete 1` +
-Deletes the 1st person in the results of the `find` command.
+* `adddiet S1234567A`
+** `d/halal d/kosher d/babyfood`
```
Since you can add multiple dietary restrictions, you should make that clear in the instruction. You can refer to the original AB4's tag example. Basically make it clear that d/dietary restriction can occur one or more times in the adddiet command

kexiaowen commented on docs/UserGuide.adoc:
```diff
@@ -56,177 +56,211 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
-=== Selecting a person : `select`
+=== View patient’s medication: UI functionality
+View a patient’s medication history.
```
Keep in mind that the focus is still on command line inputs. If possible there should be a CLI command equivalent

kexiaowen commented on docs/UserGuide.adoc:
```diff
@@ -56,177 +56,211 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-* `delete 1` +
-`undo` (reverses the `delete 1` command) +
-`redo` (reapplies the `delete 1` command) +
+* `visitorin p/Ling Zhiyu v/Sara Ann Nicholas`
```
Why is this command using patient's name and not NRIC or index? Does this mean you do not allow the application to have two persons with the same name? If there are two persons with the same name, how will your app handle this command? If you are adding restrictions (e.g. no two patients can have the same name) do remember to specify them!

kexiaowen reviewed (commented):
Good job team! User guide and developer guide are clear in general. Pointed out some minor issues that you can look into.

darienchong commented:
Hi @kexiaowen, could you review our DG? 
We've added our implementation details (as per v1.2 requirements) to the DG.
Thanks.

kexiaowen commented:
Good effort in updating the DG and adding sequence diagrams! 👍 
Some issues spotted:
1. @snajef The parse method should not be called within creating `AddmedsCommandParser`. Refer to the Delete Command sequence diagram if you are still unclear.
<img width="270" alt="screen shot 2018-10-23 at 10 49 50 pm" src="https://user-images.githubusercontent.com/20796648/47369403-2b451480-d716-11e8-8cc2-94203b40c6e0.png">
<img width="352" alt="screen shot 2018-10-23 at 10 50 26 pm" src="https://user-images.githubusercontent.com/20796648/47369411-2f713200-d716-11e8-991b-b4a0085c4d3a.png">

2. @jeffypie369 I believe this table's entries are wrong.
<img width="1028" alt="screen shot 2018-10-23 at 11 01 11 pm" src="https://user-images.githubusercontent.com/20796648/47370149-aeb33580-d717-11e8-8f87-f3d24e888ad5.png">
For the sequence diagram, same mistake as Darien. First `AddressBookParser` needs to construct a new `AddApptCommandParser`. A new `AddApptCommandParser` should be returned first before addr book parser calls AddApptCommandParser's parse method. Again you can refer to the sample sequence diagram given for Delete Command.

3. @omegafishy Label clearly the APIs you are calling. Some of the method calls are missing the method name. Please check if you made the same mistake in sequence diagram as Darien and Jefferson (Looks ok but I can't be sure as method names are missing). 

Any design considerations that you can add?

4. @yuntongzhang 
<img width="236" alt="screen shot 2018-10-23 at 11 19 41 pm" src="https://user-images.githubusercontent.com/20796648/47371320-28e4b980-d71a-11e8-877d-7ab36149054f.png">
This part of your sequence diagram is not right. Refer to my explanations above.

<img width="855" alt="screen shot 2018-10-23 at 11 21 29 pm" src="https://user-images.githubusercontent.com/20796648/47371433-65b0b080-d71a-11e8-84af-468079266255.png">
**an Enum

5. @GAOJIAXIN666 :( Did not see your updates/diagrams for your command in DG. Please try to do it asap.

GAOJIAXIN666 committed:
Update visitorin component (#110)

* change visitoIncommand and its parser class name

* change to sort by Nric instead of Name

* update visitor list structure

* add more test cases for coverage

* corre the example in visitorin command

* Rename VisitorInCommand.java to VisitorinCommand.java

change visitorinCommand file name

* Rename VisitorInCommandParser.java to VisitorinCommandParser.java

* Update VisitorListTest.java

jeffypie369 committed:
Fix bugs

enochgao-zz committed:
add viewvisitors command and its parser

jeffypie369 committed:
Add AppointmentView

jeffypie369 committed:
Add AppointmentView

jeffypie369 committed:
Fix bugs

jeffypie369 committed:
Add check nric functionality

enochgao-zz committed:
add Viewvisitorscommand test and its parser test

enochgao-zz committed:
add visitorout command and its parser

enochgao-zz committed:
add feature visitorout command, its parser and tests

enochgao-zz committed:
add a test as coverage dropped

enochgao-zz committed:
Merge branch 'feature/documentation' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/viewvisitor+visitorout

jeffypie369 committed:
Fix bugs in CheckinCommandTest, CheckinCommandSystemTest, EditCommandSystemTest for duplicate NRIC input

lingzhiyu committed:
Add SwappablePanelName.HISTORY enum

lingzhiyu committed:
Add HistoryView UI and relevant classes

lingzhiyu committed:
Update MainWindow to include HistoryView functionality

enochgao-zz committed:
add developer guide for visitorin, viewvisitors, visitorout part

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

lingzhiyu committed:
Add XmlAdaptedDiagnosis

lingzhiyu committed:
Add XmlAdaptedMedicalHistory

yuntongzhang committed:
Add Xml component for adddiet command (#118)

* Add xml adapted version of diet and dietCollection

* Add more tests for Diet class

* Add method to DietCollection and update test

* Fix failing Diet tests

* Integrate diet component to XmlAdaptedPerson

* Rename some of the tests in DietTest

* Add test for XmlAdaptedDiet

* Add test for XmlAdaptedDietCollection

* Fix failing tests and checkstyle

enochgao-zz committed:
add sequence diagram for visitors related commands in developer guide

lingzhiyu committed:
Update MedicalHistory class

lingzhiyu committed:
Update HistoryPanel.fxml

lingzhiyu committed:
Fix minor redundant lines

lingzhiyu committed:
Add doctor regex into Diagnosis

lingzhiyu committed:
Updated Person due to changes in MedicalHistory implementation

lingzhiyu committed:
Update getSampleMedicalHistory method

lingzhiyu committed:
Update PersonBuilder testutil

lingzhiyu committed:
Update ViewmhCommandTest due to changes in MedicalHistory implementation

lingzhiyu committed:
Update PersonTest

lingzhiyu committed:
Add storage file junit tests

lingzhiyu committed:
Add HistoryViewTest

enochgao-zz committed:
update user guide for visitor's commands

lingzhiyu committed:
Fix minor bug in Addmeds, to include MedicalHistory of existing patient

lingzhiyu committed:
Update XmlAdaptedPerson to include XmlAdaptedMedicalHistory

lingzhiyu committed:
Update XmlAdaptedDiagnosis and XmlAdaptedMedicalHistory with minor
changes

yuntongzhang committed:
Add JavaFx related methods to DietCollection class

yuntongzhang committed:
Add new SwappablePanelName and relevant tests for diets

darienchong committed:
Add Project Portfolio Page for Darien (#106)

* Update sequence diagram for `addmeds` to fit the style of the other sequence diagrams
Increase width of sequence diagram image in developer guide

* Update User Guide
Sort commands in User Guide in lexicographical order

* Label all images in User Guide by figure number.
Add references to figure numbers in User Guide.

* Add newline at EOF of UserGuide.adoc to fit checkstyle requirements

* Change format for figure indication

* Standardise capitalisation for "component" and "implementation" across developer guide.
Add figure numbers for images
Standardise capitalisation across title headings
Update class diagrams

* Created my project portfolio page

* Document `select` command in User Guide

darienchong committed:
Refactor addmeds and viewmeds (#112)

* Refactor addmeds and nric to provide a suitable error message in the Ui when a non-conforming NRIC is presented.

* Refactor MainWindow to follow SLAP

* Refactor MedicationView to use a more natural method of mapping column indices to columns.

* Fix a minor formatting issue

jeffypie369 committed:
Correct PPP

jeffypie369 committed:
Fix bug

yuntongzhang committed:
Add fxml file for Diet panel

jeffypie369 committed:
Fix bugs

lingzhiyu committed:
Update HistoryPanel sizes

lingzhiyu committed:
Update application title

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main into two

jeffypie369 committed:
Update AppointmentView

jeffypie369 committed:
Merge pull request #9 from jeffypie369/two

Update AppointmentView

lingzhiyu committed:
Fix checkstyle bugs

jeffypie369 committed:
Add coverage

jeffypie369 committed:
Merge pull request #10 from jeffypie369/two

Add coverage

lingzhiyu committed:
Merge pull request #121 from omegafishy/update_diagnosis

Add Timestamp + update Diagnosis and profile picture

jeffypie369 committed:
Add coverage

jeffypie369 committed:
Merge pull request #11 from jeffypie369/two

Add coverage

lingzhiyu committed:
Update MainWindowTest

lingzhiyu committed:
Merge pull request #119 from jeffypie369/one

Add AppointmentView and check nric functionality

lingzhiyu committed:
Update Timestamp documentation

jeffypie369 committed:
Merge branch 'master' into master

lingzhiyu committed:
Merge pull request #114 from jeffypie369/master

Add PPP and AppointmentView without tests

lingzhiyu committed:
Merge pull request #116 from CS2103-AY1819S1-W14-3/feature/viewvisitor+visitorout

Feature/viewvisitor+visitorout

lingzhiyu committed:
Resolve checkstyle errors

yuntongzhang committed:
Add DietView class

yuntongzhang committed:
Integrate DietView into MainWindow

lingzhiyu committed:
Merge pull request #120 from omegafishy/v1.3_production

Add addmh UI functionality

yuntongzhang committed:
Add tests for DietView

yuntongzhang committed:
Fix failing tests and checkstyle

yuntongzhang committed:
Update app version number

yuntongzhang committed:
Add DietCollection to PersonBuilder

yuntongzhang committed:
Fix failing test

yuntongzhang committed:
Merge pull request #122 from yuntongzhang/diet-view

Add views for dietary requirements.

darienchong committed:
Setup framework for new developer-only command to generate demo data.
Recommend command to be disabled before release.

darienchong committed:
Fill template data for generation

darienchong committed:
Add more example data

darienchong committed:
Fix a bug where the list of duplicates wasn't being cleared after generating.

darienchong committed:
Fix checkstyle violations.

jeffypie369 committed:
Add check for date and time input

jeffypie369 committed:
Add coverage for date-time exception;

jeffypie369 committed:
Add check for date and time input (#125)

* Add check for date and time input

* Add coverage for date-time exception;

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

darienchong committed:
Fix a bug where a few columns in the Medication view were being sorted by lexicographical instead of numerical order.

lingzhiyu committed:
Merge pull request #127 from CS2103-AY1819S1-W14-3/feature/sort+view

Feature/sort+view

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/project-portfolio-page

yuntongzhang committed:
Add ppp for yuntongzhang (#128)

jeffypie369 committed:
Update PPP

darienchong committed:
Update documentation according to comments from Diana in 01-11-18 CS2101 session.

darienchong committed:
Rename HMS2K18 to HealthBase

darienchong committed:
Fix documentation according to CS2101 standard (#130)

* Created my project portfolio page

* Update documentation according to comments from Diana in 01-11-18 CS2101 session.

* Rename HMS2K18 to HealthBase

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/generate-data

lingzhiyu committed:
Refactored application name to HealthBase, did not include UGDG (pending)

lingzhiyu committed:
Update UGDG, add omegafishy PPP

lingzhiyu committed:
Merge pull request #132 from omegafishy/v1.4_production

Refactored application name to HealthBase, update documentation

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/generate-data

darienchong committed:
Finish up generator code
Add tests for GendataCommand and GendataCommandParser
Add documentation for new command in Developer Guide (as the command is not meant for typical user use, its documentation was omitted from the User Guide)

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/project-portfolio-page

# Conflicts:
#	docs/UserGuide.adoc

jeffypie369 committed:
Fix Type bug

jeffypie369 committed:
Fix more addappt command exception bugs

darienchong committed:
Fix bug highlighted in PE dry run.

lingzhiyu committed:
Update asciidoc website header

lingzhiyu committed:
Merge pull request #170 from omegafishy/v1.4_production

Update asciidoc website header

lingzhiyu committed:
Fix bug from AddmhCommand

darienchong committed:
Fix wrong example usage of `view` command. (#169)

yuntongzhang committed:
Minor fixes for yuntongzhang ppp (#129)

* Fix the ending tag in user guide

* Modify links in yuntongzhang ppp

* docs/AboutUs: update app name, add link to yuntongzhang ppp

darienchong committed:
Minor fixes for snajef ppp (#141)

* Created my project portfolio page

* Update documentation according to comments from Diana in 01-11-18 CS2101 session.

* Rename HMS2K18 to HealthBase

yuntongzhang committed:
Add a list of checkedOutPerson in the model manager

yuntongzhang committed:
Remove redundant fielf in ModelManager

yuntongzhang committed:
Support checkedOutPerson in the XML storage

yuntongzhang committed:
Minor refactor of old AddDietCommand

yuntongzhang committed:
Add new CheckoutCommand functionality

yuntongzhang committed:
Change tag for the XML person elements

yuntongzhang committed:
Finish up tests for adddiet command

yuntongzhang committed:
Fix failing tests because of change in model and address book

yuntongzhang committed:
Add tests for changes in AddressBook and ModelManager

yuntongzhang committed:
Fix checkstyle

yuntongzhang committed:
Add tests for CheckoutCommand and parser

yuntongzhang committed:
Refactor checkin command to register command

yuntongzhang committed:
Clean up related docs for old checkin command

yuntongzhang committed:
Add new checkin command functionality

lingzhiyu committed:
Merge pull request #172 from omegafishy/v1.4_production

Fix bug from AddmhCommand

lingzhiyu committed:
Add XmlAdaptedTimestamp.
Update XmlAdaptedDiagnosis and Timestamp accordingly.

lingzhiyu committed:
Update XmlAdaptedDiagnosis

yuntongzhang committed:
Add tests for the new CheckinCommand and parser

yuntongzhang committed:
Merge pull request #140 from CS2103-AY1819S1-W14-3/feature/generate-data

Feature/generate data

yuntongzhang committed:
Merge pull request #173 from CS2103-AY1819S1-W14-3/checkout-patient

Add functionality to checkout patient and checkin returning patient

lingzhiyu committed:
Add DiagnosisTest to increase coverage

lingzhiyu committed:
Remove application code bindings to ViewmhCommand.

lingzhiyu committed:
Resolve todo comments

jeffypie369 committed:
Merge branch 'master' into master

jeffypie369 committed:
Change doctor regex

jeffypie369 committed:
Merge branch 'master' of https://github.com/jeffypie369/main

jeffypie369 committed:
Fix Type type

jeffypie369 committed:
Fix addappt date and time bugs, check for duplicates and validity

darienchong committed:
Fix bugs related to medication (#168)

* Refactor addmeds and nric to provide a suitable error message in the Ui when a non-conforming NRIC is presented.

* Refactor MainWindow to follow SLAP

* Refactor MedicationView to use a more natural method of mapping column indices to columns.

* Fix a minor formatting issue

* Address bugs brought up in PE dry run

jeffypie369 committed:
Fix bugs related to type, date and time (#178)

* Add check for date and time input

* Add coverage for date-time exception;

* Update PPP

* Fix Type bug

* Fix more addappt command exception bugs

* Fix bugs

* Change doctor regex

* Fix Type type

* Fix addappt date and time bugs, check for duplicates and validity

lingzhiyu committed:
Update UGDG, PPP to generate updated gh pages  (#181)

* Update user and developer guide

* Update user and developer guide

kexiaowen commented:
Hi @jeffypie369 @yuntongzhang @GAOJIAXIN666 Just took a look at your sequence diagrams and it seems like the issue about calling the parse method within constructing a command parser still persists. Please try to update it asap. You can refer to @snajef 's updated sequence diagram for reference.

jeffypie369 committed:
Fix trailing white space and var name

jeffypie369 committed:
Fix addappt NRIC bug

jeffypie369 committed:
Add test for invalid date and time in addappt

jeffypie369 committed:
Add test for invalid type in addappt

lingzhiyu committed:
Attempt to fix DateTimeParseException error in TravisCI test

jeffypie369 committed:
Add tests for invalid inputs for addappt

jeffypie369 committed:
Merge branch 'master' into bugfixes

lingzhiyu committed:
Merge pull request #182 from jeffypie369/bugfixes

Bug fixes and tests added

lingzhiyu committed:
Attempt to fix IllegalParseException error in TravisCI test

jeffypie369 committed:
Edit AddapptSequenceDiagram

jeffypie369 committed:
Update UGDG

jeffypie369 committed:
Edit UG

jeffypie369 committed:
Edit UG

jeffypie369 committed:
Fix incorrect instruction

jeffypie369 committed:
Fix incorrect register instruction

jeffypie369 committed:
Fix drug allergy output bug (#185)

* Add test for invalid nric for addappt

* Remove nric test for addappt

* Fix drug allergy output bug

jeffypie369 committed:
Make changes to UGDG by Jeff (#186)

* Edit AddapptSequenceDiagram

* Update UGDG

* Edit UG

* Edit UG

* Fix incorrect instruction

* Fix incorrect register instruction

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

jeffypie369 committed:
Fix test for addappt

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/project-portfolio-page

# Conflicts:
#	docs/UserGuide.adoc

darienchong committed:
Edit PPP to fit 2101 requirements

darienchong committed:
Feature/project portfolio page for darien (#188)

* Created my project portfolio page

* Update documentation according to comments from Diana in 01-11-18 CS2101 session.

* Rename HMS2K18 to HealthBase

* Edit PPP to fit 2101 requirements

jeffypie369 committed:
Rearrage command orders of DG

lingzhiyu committed:
Merge pull request #180 from omegafishy/v1.4_production

Clean up code, add extra test

yuntongzhang committed:
Update UGDG for yuntongzhang (#183)

* Update UserGuide for checkin checkout viewdiets

* Update sequence according to tutor's comment

jeffypie369 committed:
Fix updatePerson test

yuntongzhang committed:
Fix formatting of UserGuide (#190)

lingzhiyu committed:
Merge pull request #187 from jeffypie369/master

Fix test and rearrange DG

jeffypie369 committed:
Fix formatting of register command

jeffypie369 committed:
Fix typo

jeffypie369 committed:
Add restriction for addition of drug allergies

jeffypie369 committed:
Remove redundant commands in UG

jeffypie369 committed:
Change sample drug allergies

lingzhiyu committed:
Attempt to fix TravisCI build error. Set locale according to machine's default.

darienchong committed:
Feature/refactor person (#189)

* Refactor Person class to make constructing a new person less tedious, also more extensible for future additions

* Refactor Person to be more extensible

jeffypie369 committed:
Improve UserGuide for register command and fix default tags (#197)

* Fix formatting of register command

* Fix typo

* Add restriction for addition of drug allergies

* Remove redundant commands in UG

* Change sample drug allergies

lingzhiyu committed:
Add travisci error message verbosity

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

yuntongzhang committed:
Defensive programming of DietCollection (#192)

* Defensive programming of DietCollection

* Remove constructor as it is included in another PR

* Add back defensive copy constructor

darienchong committed:
Change filename of "HighlightIndex.png" to "highlightIndex.png" so that the user guide displays the image correctly. (#199)

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

lingzhiyu committed:
Change LocalDateTime locale to en_SG

lingzhiyu committed:
Merge pull request #177 from omegafishy/bug_fixes

Add XmlAdaptedTimestamp

lingzhiyu committed:
Update AddmhCommand and Timestamp, correct documentation and message constants

darienchong committed:
Change time intervals for generated appointments to be half hour intervals (#206)

Add generation of visitors

jeffypie369 committed:
Merge branch 'master' of https://github.com/nus-cs2103t-w14-3/main

jeffypie369 committed:
Remove DateTime class

jeffypie369 committed:
Remove DateTime class

yuntongzhang committed:
Refactor `*Command#execute` and equality definition of `Person` (#209)

* Fix typo in UserGuide

* Modify equality check for Person, Update register error message

* Refactor *Command#execute and improve error message

* Fix checkstyle

* Improve checkout UserGuide

* Minor fix on RegisterCommand error message

jeffypie369 committed:
Refactor AddApptCommand

jeffypie369 committed:
Refactor AddApptCommand and AddApptCommandParser

jeffypie369 committed:
Merge branch 'master' into appointment

jeffypie369 committed:
Fix trailing whitespace

jeffypie369 committed:
Fix merge conflict

jeffypie369 committed:
Fix failing test

darienchong committed:
Feature/addmeds+viewmeds (#205)

* Refactor addmeds and nric to provide a suitable error message in the Ui when a non-conforming NRIC is presented.

* Refactor MainWindow to follow SLAP

* Refactor MedicationView to use a more natural method of mapping column indices to columns.

* Fix a minor formatting issue

* Address bugs brought up in PE dry run

* Add checks to ensure drug name and dosage unit are not empty.

* Fix (?) trailing whitespace in AddmedsCommandParser.java
Was unable to find this trailing whitespace, committing to force Travis to rebuild.

* Update AddmedsCommandParser.java

Remove trailing whitespace.

* Remove unused imports in AddmedsCommand
Remove trailing whitespace in AddmedsCommandParser

jeffypie369 committed:
Fix parse nric bug in addappt

darienchong committed:
Feature/documentation (#212)

* Change filename of "HighlightIndex.png" to "highlightIndex.png" so that the user guide displays the image correctly.

* Update user guide to reflect changes in v1.3

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/project-portfolio-page

darienchong committed:
Update PPP

lingzhiyu committed:
Merge pull request #207 from jeffypie369/appointment

Refactor Appointment

darienchong committed:
Feature/project portfolio page (#213)

* Created my project portfolio page

* Update documentation according to comments from Diana in 01-11-18 CS2101 session.

* Rename HMS2K18 to HealthBase

* Edit PPP to fit 2101 requirements

* Update PPP

yuntongzhang committed:
Add input validation to adddiet command (#214)

* Add input validation for adddiet

* Update UserGuide for adddiet command

lingzhiyu committed:
Update AddmhCommand and Timestamp, to improve javadocs and update AddmhCommand messages

lingzhiyu committed:
Update UGDG and PPP for omegafishy

lingzhiyu committed:
Update DG

lingzhiyu committed:
Update AddmhCommand, remove obsolete method

darienchong committed:
Fix addmeds diagram in Developer Guide

darienchong committed:
Add missing <<static>> tag in method call of addmeds sequence diagram in Developer Guide

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/project-portfolio-page

yuntongzhang committed:
Set up Reposense configuration (#221)

* Config Reposense

* Add more names in Reposense

* Fix checkstyle

lingzhiyu committed:
Update AddmhSequenceDiagram.png

lingzhiyu committed:
Update diagram width

jeffypie369 committed:
Refactor AddressBook to HealthBase (#218)

lingzhiyu committed:
Update UGDG and PPP

lingzhiyu committed:
Update UGDG and PPP

darienchong committed:
Update sequence diagram to include CommandUtil#getPatient interactions with model.

lingzhiyu committed:
Update UGDG and PPP

darienchong committed:
Shorten PPP to fit 10-page requirement.

lingzhiyu committed:
Merge branch 'master' into update_docs

GAOJIAXIN666 committed:
Feature/viewvisitor+visitorout (#208)

* Created my project portfolio page

* ppp for portfolio

* Update gaojiaxin.adoc

* Update gaojiaxin.adoc

* update local with remote main

* add xml related classes for visitor and ui for visitor

* update XmlAdapted visitor&visitorlist related tests

* add more test cases in xmladapted visitor and visitorlist

* Update AddmedsCommand.java

* update developerguide and userguide for visitors parts

* update userguide for visitors parts

* update ppp

* add VisitorViewTest

* Update XmlAdaptedVisitorList.java

* Update XmlAdaptedVisitorTest.java

* update xmlAdaptedVisitorListTest

* Update XmlAdaptedVisitor.java

* Update VisitorList.java

* update DeveloperGuide and UserGuide

lingzhiyu committed:
Merge branch 'master' into update_docs

lingzhiyu committed:
Merge pull request #220 from CS2103-AY1819S1-W14-3/feature/project-portfolio-page

Update `addmeds` sequence diagram, Update PPP

lingzhiyu committed:
Merge pull request #184 from omegafishy/update_docs

Update UGDG for omegafishy PPP

darienchong committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-W14-3/main into feature/sort+view

yuntongzhang committed:
Refactor code (#222)

* Refactor logic component except parser

* Refactor model component and some comments on logic

* Refactor storage component

* Refactor ui component

* Refactor tests

darienchong committed:
Fix outdated comment in ViewCommandParser

darienchong committed:
Minor comment fixes for outdated comments in ViewCommandParser and SortCommandParserTest

darienchong committed:
Fix outdated comment in MedicationViewHandle

lingzhiyu committed:
Merge pull request #224 from CS2103-AY1819S1-W14-3/feature/sort+view

Fix typos in comments

jeffypie369 committed:
Jeff's UG, DG, PPP (#215)

* Make changes to UG and PPP

* Make changes to UG and PPP

* Make changes to DG

* Refactor AddApptCommand to better suit Sequence Diagram

* Update sequence of addappt in DG

* Fix sequence of addappt in DG

* Edit AddapptSequenceDiagram

* Fix DG

* Edit ppp

* Update UGDG

* Add screenshots for UG

* Update AddapptSequenceDiagram

* Add addapptactivitydiagram

* Make changes to PPP

* Update DG

* resolve conflict

* Add test instructions in DG

* Edit UG

* Edit UGDGPPP

darienchong committed:
Add manual testing instructions to DG (#223)

* Fix bug highlighted in PE dry run.

* Change filename of "HighlightIndex.png" to "highlightIndex.png" so that the user guide displays the image correctly.

* Update user guide to reflect changes in v1.3

* Add manual testing instructions for addmeds, view, sort, and gendata.

* Removing trailing whitespace.

* Fix outdated comment in ViewCommandParser

* Minor comment fixes for outdated comments in ViewCommandParser and SortCommandParserTest

* Fix outdated comment in MedicationViewHandle

* Fix typo

lingzhiyu committed:
Add activity, sequence and some PPP diagrams.

lingzhiyu committed:
Update UGDG, include manual testing for developer guide

lingzhiyu committed:
Removed incorrectly pushed diagram from /build.

jeffypie369 committed:
Edit code snippets (#226)

* Edit code snippets

* Update UG

lingzhiyu committed:
Updated UGDG, PPP and diagrams

darienchong committed:
Update `addmeds` sequence diagram (#228)

* Update sequence diagram to replace references to "AddressBook" with "HealthBase"

* Add code contributions

jeffypie369 committed:
Update profile (#231)

lingzhiyu committed:
Updated DG and PPP

GAOJIAXIN666 committed:
Gaojiaxin ppp (#230)

* Update gaojiaxin.adoc

* Update gaojiaxin.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update gaojiaxin.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

yuntongzhang committed:
Yuntong UG DG PPP (#225)

* Update yuntongzhang PPP

* Update sequence diagram

* Add testing instructions

* Minor fixes

* Update sequence diagram

* Add diagram in UserGuide

jeffypie369 committed:
(Hopefully) Final PPP (#232)

* Update profile

* Update PPP

lingzhiyu committed:
Updated PPP and AboutUs

lingzhiyu committed:
Add EOF

lingzhiyu committed:
Merge pull request #227 from omegafishy/update_docs

Update omegafishy PPP, UGDG

yuntongzhang committed:
Fix grammar in DeveloperGuide (#233)

* Fix grammar in DeveloperGuide

* Fix wording in testing

yuntongzhang committed:
Update version number (#234)

lingzhiyu committed:
Update AboutUs

yuntongzhang committed:
fix guide (#235)

lingzhiyu committed:
Delete Ui

lingzhiyu committed:
Add updated Ui

jeffypie369 committed:
Fix failing screenshots in UserGuide (#239)

* Update profile

* Change jpg to png

yuntongzhang committed:
Minor fixes on DeveloperGuide and comments (#240)

Changes made:

1. Delete a few obsolete lines.
2. Changed Address Book to HealthBase in the text and some diagrams.
3. Reordered manual testing sections.
